pandas/core/dtypes/cast.py

```diff
@@ -154,6 +155,30 @@ def maybe_box_datetimelike(value: Scalar, dtype: Optional[Dtype] = None) -> Scal
     return value
 
 
+def maybe_box_native(value: Scalar) -> Scalar:
+    """
+    If passed a scalar cast the scalar to a python native type.
+
+    Parameters
+    ----------
+    value : scalar or Series
+
+    Returns
+    -------
+    scalar or Series
+    """
+    if is_datetime_or_timedelta_dtype(value):
+        value = maybe_box_datetimelike(value)
+    elif is_float_dtype(value):
+        value = float(value)
+    elif is_integer_dtype(value):
+        with suppress(ValueError, TypeError):
```

Comment:
whats an example that fails here?

Reply:
I've pushed a commit removing this. I recall I put this here to fix a bug (but can't remember what - will see from CI failures, if any)

Reply:
looks like it's not needed